src/Mvc/Mvc.RazorPages/src/Builder/RazorPagesEndpointRouteBuilderExtensions.cs

```diff
@@ -31,6 +32,245 @@ public static IEndpointConventionBuilder MapRazorPages(this IEndpointRouteBuilde
             return GetOrCreateDataSource(routes);
         }
 
+        /// <summary>
+        /// Adds a specialized <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that will match
+        /// requests for non-filenames with a very low priority. The request will be routed to a page endpoint that
+        /// matches <paramref name="page"/>.
```

Comment:
I'm wondering whether it might be better *not* to have the various `MapFallbackTo*` overloads that don't take a `pattern`. Those ones all "match requests for non-filenames", but TBH that feels like a pretty arbitrary specialization. I know it matches our current use cases, but it feels like a separate concern from "mapping a fallback route".

It's not like developers will be writing lots of calls to `MapFallbackTo*`. Chances are, they will have exactly one such call in their whole application, and in most of those cases it was put there for them by the template anyway.

If the template produced more explicit calls, e.g.,

    routes.MapFallbackToPage("Index.cshtml", "{*path:nonfile}");

... then it would be more obvious that the `nonfile` constraint is being applied, plus how to remove it if you don't want it.

Reply:
I mostly agree and it would simplify the docs. I tend to prefer these kinds of things being explicit anyway. What I don't want to is have to put these simpler overloads back as the result of a template review.

Reply:
I mocked this up here: https://gist.github.com/rynowak/ab9305b5f1dbb2d918e568b4129a8829

I can't really see us putting this route in a template because it's a lot to explain. I'm going to leave the simpler overloads in place for now, and we can review and decide later.

Reply:
Alright, I agree.